service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpCLIService.java

```diff
@@ -94,7 +95,13 @@ protected void initServer() {
       String threadPoolName = "HiveServer2-HttpHandler-Pool";
       ThreadPoolExecutor executorService = new ThreadPoolExecutor(minWorkerThreads,
           maxWorkerThreads,workerKeepAliveTime, TimeUnit.SECONDS,
-          new SynchronousQueue<Runnable>(), new ThreadFactoryWithGarbageCleanup(threadPoolName));
+          new SynchronousQueue<Runnable>(), new ThreadFactoryWithGarbageCleanup(threadPoolName)) {
+        @Override
+        public void setThreadFactory(ThreadFactory threadFactory) {
```

Comment:
In this case would no be better to mark this method as an abstract method? Or have a parent that is abstract? To have an empty method that will be eventually overridden is bit strange to me. Maybe I don't have a context, but it seem strange :)

Reply:
@dengzhhu653 should it be like
````
      ThreadPoolExecutor executorService = new ThreadPoolExecutor(minWorkerThreads,
          maxWorkerThreads,workerKeepAliveTime, TimeUnit.SECONDS,
        new SynchronousQueue<>(), new ThreadFactoryWithGarbageCleanup(threadPoolName)){
            @Override
            public void setThreadFactory(ThreadFactory threadFactory) {
              if (getThreadFactory() != null) {
                LOG.warn("Ignore setting the thread factory as the pool has already provided his own: {}", getThreadFactory());
              } else {
                super.setThreadFactory(threadFactory);
              }
            }
      };
````